Composer/packages/client/src/components/Auth/TenantDialog.tsx

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import { DialogWrapper, DialogTypes } from '@bfc/ui-shared';
+import formatMessage from 'format-message';
+import { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import { ChoiceGroup } from 'office-ui-fabric-react/lib/ChoiceGroup';
+import { useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import { availableTenantsState, dispatcherState } from '../../recoilModel/atoms';
+
+export interface TenantDialogProps {
+  onDismiss: () => void;
+}
+export const TenantDialog: React.FC<TenantDialogProps> = (props) => {
+  const availableTenants = useRecoilValue(availableTenantsState);
+  const { setCurrentTenant } = useRecoilValue(dispatcherState);
+  const [tenant, setTenant] = useState<string>('');
+
+  return (
+    <DialogWrapper
+      isBlocking
+      isOpen
+      dialogType={DialogTypes.CreateFlow}
+      subText={formatMessage(
+        'The directory you choose will impact subscription, resource group, and region filters that are available when you select or create Azure resources. You can change this later.'
+      )}
+      title={formatMessage('Select directory')}
+      onDismiss={props.onDismiss}
+    >
+      <ChoiceGroup
```

Comment:
isn't this supposed to show the selected tenant?

Reply:
meaning pre-populate the previously chosen tenant?

Reply:
that's right

Reply:
I will check with design when I can -- typically, we only show the tenant selection when we do not know or have a previous tenant. However, in the case where user is changing tenants we may want to display the current selection.

Reply:
ok this has been updated with feedback from @emivers8  - the current and/or default tenant will be pre-selected.